app/Models/Campaign.php

```diff
@@ -267,4 +269,14 @@ public function setArbitraryDateString($attribute, $value)
 
         $this->attributes[$attribute] = $this->fromDateTime($value);
     }
+
+    /**
+     * Get the indexable data array for the model.
+     *
+     * @return array
+     */
+    public function toSearchableArray()
```

Comment:
How does this work? Is Scout just auto watching things etc. since we implement the `Searchable ` module, and we're just extending our own customized array of data which they pull?

Reply:
It's adds a model observer that triggers whenever any activity happens on the model utilizing it. So when creating/updating/deleting etc, the observer is triggered :)